### PR TITLE
[REVIEW] Fix random string data length to become variable

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR resolves an issue where all random strings generated in a series are of the same length. With this change rows of a string column will have random length data.

this PR:
```python
In [1]: from cudf.testing.dataset_generator import rand_dataframe


In [2]: cudf.DataFrame.from_arrow(rand_dataframe(
        [
            {
                "dtype": "str",
                "null_frequency": 0,
                "cardinality": 100,
                "max_string_length": 10,
            },
        ],
        rows=10,
    ))['1']
Out[2]: 
0                                          

1                                     }\1\nl
2                              vd|byshzO<R*4
3                           TmA5@%BK
WWXow`,
4              #WK
p1W\rTK3.. %}XnXmf>%d`SKp
5                                     a`N!Ua
6                                           
7       i` n6Ma}WthGa-Rd}wL@_O`i<=@|#YL E:X:
8                      
40m\r#sC_CYMibXEEiot
9    
NTn6Mb<z^g}Oizs?|S:0i}Vw3
e48
t\~\nD]G
dtype: object
```

`branch-22.06`:
```python
In [1]: from cudf.testing.dataset_generator import rand_dataframe


In [2]: cudf.DataFrame.from_arrow(rand_dataframe(
        [
            {
                "dtype": "str",
                "null_frequency": 0,
                "cardinality": 100,
                "max_string_length": 10,
            },
        ],
        rows=10,
    ))['1']
Out[2]: 
0    M9FD7xIOaT
1    CE0yJmYXc1
2    uTtn2R8XcI
3    boVUoYjOMD
4    3UCdG47tSI
5    3SgM5ONync
6    JsrChyuULX
7    sbTsyA3WjO
8    Xff9TcZVWn
9    6NxIyLGKCI
dtype: object

```

### CODE REVIEW

File: python/cudf/cudf/testing/_utils.py [hunk 1]
```diff
@@ -2,12 +2,14 @@
 
 import itertools
 import re
+import string
```
Comment (galipremsagar) [inline python/cudf/cudf/testing/_utils.py:5]:
```suggestion
```

File: python/cudf/cudf/testing/_utils.py [hunk 2]
```diff
@@ -2,12 +2,14 @@
...
 import warnings
 from collections import abc
 from contextlib import contextmanager
 from decimal import Decimal
 
 import cupy
+import mimesis
```
Comment (galipremsagar) [inline python/cudf/cudf/testing/_utils.py:12]:
```suggestion
```

File: python/cudf/cudf/testing/_utils.py [hunk 3]
```diff
@@ -306,7 +308,15 @@ def gen_rand(dtype, size, **kwargs):
...
-        return pd.util.testing.rands_array(10, size)
+        return np.array(
+            [
+                mimesis.random.random.schoice(
+                    string.printable,
+                    np.random.randint(low=0, high=50, size=1)[0],
+                )
+                for _ in range(size)
+            ]
+        )
```
Comment (galipremsagar) [inline python/cudf/cudf/testing/_utils.py:319]:
```suggestion
        return pd.util.testing.rands_array(10, size)
```

**Review by brandon-b-miller — APPROVED** [review]

**Comment by brandon-b-miller** [comment]
Is there a quick test that could go in with this?

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**